packages/headless/src/api/analytics/insight-analytics.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  AnalyticsClientSendEventHook,
+  InsightClientProvider,
+  CoveoInsightClient,
+  history,
+} from 'coveo.analytics';
+import {SearchEventRequest} from 'coveo.analytics/dist/definitions/events';
+import {Logger} from 'pino';
+import {
+  buildFacetStateMetadata,
+  getStateNeededForFacetMetadata,
+  SectionNeededForFacetMetadata,
+} from '../../features/facets/facet-set/facet-set-analytics-actions-utils';
+import {getQueryInitialState} from '../../features/query/query-state';
+import {getSearchInitialState} from '../../features/search/search-state';
+import {
+  ConfigurationSection,
+  PipelineSection,
+  QuerySection,
+  SearchHubSection,
+  SearchSection,
+} from '../../state/state-sections';
+import {PreprocessRequest} from '../preprocess-request';
+import {BaseAnalyticsProvider} from './base-analytics';
+
+export type StateNeededByInsightAnalyticsProvider = ConfigurationSection &
+  Partial<
+    SearchHubSection &
+      SearchSection &
+      PipelineSection &
+      QuerySection &
+      SectionNeededForFacetMetadata
+  >;
+
+export class InsightAnalyticsProvider
+  extends BaseAnalyticsProvider
+  implements InsightClientProvider
+{
+  public getSearchUID(): string {
+    return (
+      this.state.search?.searchResponseId ||
+      this.state.search?.response.searchUid ||
+      getSearchInitialState().response.searchUid
+    );
+  }
+  public getPipeline(): string {
+    return (
+      this.state.pipeline || this.state.search?.response.pipeline || 'default'
```

Comment:
I don't think it's true that the 'default' pipeline is always the "default" :) someone could make their default pipeline named 'foo' if they wanted.

When would we need those fallbacks where both the state and the last response state are empty exactly?